hockey-score/src/app/shared/header/header.component.ts

```diff
@@ -7,18 +8,19 @@ import { Component, OnInit } from '@angular/core';
 })
 export class HeaderComponent implements OnInit {
 
-  counter: number = 0;
-  constructor() { }
+  isLogin: boolean;
+
+  constructor(private readonly authService: AuthService) {
+    this.isLogin = false;
+    this.authService.isLogin.subscribe(result => this.isLogin = result);
+  }
 
   ngOnInit() {
+    this.isLogin = false;
```

Comment:
this.authService.isLogin.subscribe(result => this.isLogin = result);

you should subscribe in ngOnInit method - it is called once component not an object of type HeaderComponent (class) is created.

and to avoid memory leaks you should unsubscribe from Observable once the component is destroyed
you can implement it in the ngOnDestroy method